### PR TITLE
Add support in google_issue_tracker for Component IDs

### PR DESCRIPTION
**Pull request by rmistry**
Predator indirectly gets list of components from DIR_METADATA files in Chromium. Those files currently contain the Monorail component path (eg: "Blink>JavaScript>Compiler>Sparkplug"), after the migration they will be updated to be Buganizer component IDs (eg: "1456567").
This PR updates the google_issue_tracker code to handle these component IDs.
More context is in b/318504348

### CODE REVIEW

**Comment by rmistry** [comment]
PTAL @jonathanmetzman and @oliverchang 
CC @micahbales

**Comment by rmistry** [comment]
Checks have passed (one day we should add me as a contributor so that I can run /gcbrun).
Please merge/deploy when you can. TIA!

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py [hunk 1]
```diff
@@ -799,6 +817,37 @@ def new_issue(self):
     }
     return Issue(data, True, self)
 
+  def get_relative_component_path(self, component_id):
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:820]: Can we make this private? i.e. name this `_get_relative_component_path`.

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py [hunk 2]
```diff
@@ -799,6 +817,37 @@ def new_issue(self):
...
+      component = self._execute(
+          self.client.components().get(componentId=str(component_id)))
+    except IssueTrackerError as e:
+      if isinstance(e, IssueTrackerNotFoundError):
+        return None
+      logs.log_error('Failed to retrieve component.', component_id=component_id)
+      return None
+
+    if component['componentId'] == str(self._default_component_id):
+      return ''
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:841]: return `None` for consistency?

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py [hunk 3]
```diff
@@ -799,6 +817,37 @@ def new_issue(self):
...
+    if component.get('parentComponentId'):
+      parent_component_id = component['parentComponentId']
+      component_name = component.get('name', '')
+      if parent_component_id == str(self._default_component_id):
+        return component_name
+      return self.get_relative_component_path(
+          parent_component_id) + ">" + component_name
+    return ''
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py:849]: return `None` for consistency?

**Review by oliverchang — COMMENTED** [review]
LGTM with minor comments

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py (hunk 1 above)
Comment (rmistry) [reply src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py]: Done.

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py (hunk 2 above)
Comment (rmistry) [reply src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py]: Done.

File: src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py (hunk 3 above)
Comment (rmistry) [reply src/clusterfuzz/_internal/issue_management/google_issue_tracker/issue_tracker.py]: Done.

**Comment by rmistry** [comment]
Addressed the comments and tests pass. @jonathanmetzman , @oliverchang please merge and deploy at your convenience.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**